dev/archery/archery/crossbow/reports.py

```diff
@@ -121,6 +124,32 @@ def show(self, outstream, asset_callback=None):
                                    asset))
 
 
+class JsonReport(Report):
+
+    def url(self, query):
+        repo_url = self.job.queue.remote_url.strip('.git')
+        return '{}/branches/all?query={}'.format(repo_url, query)
+
+    def today_str(self):
+        date = datetime.utcnow()
+        return "{}-{}-{}".format(date.year, date.month, date.day)
+
+    def tasks_to_dict(self, date, tasks):
+        json_tasks = []
+        for task_name, task in tasks.items():
+            json_tasks.append({
+                "build": task_name,
+                "link": self.url(task.branch),
+                "status": task.status().combined_state.upper(),
+                "timestamp": date})
+
+        return json_tasks
+
+    def get_json_tasks(self):
+        tasks = self.tasks_to_dict(self.today_str(), self.job.tasks.items())
+        json_str = json.dump(tasks)
```

Comment:
I think this wants to be `json.dumps(tasks)` as `json.dump` wants a target file, while your goal here is to return as tring.

Reply:
i don't understand this part

Reply:
```
>>> tasks = [{
...     "build": "TASK_NAME",
...     "link": "HTTP://LINK",
...     "status": "STATUS",
...     "timestamp": "DATE"
... }]
>>> import json
>>> json.dump(tasks)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: dump() missing 1 required positional argument: 'fp'
```

while instead...

```
>>> json.dumps(tasks)
'[{"build": "TASK_NAME", "link": "HTTP://LINK", "status": "STATUS", "timestamp": "DATE"}]'

```

Reply:
thanks, it took me a while to get this